ParticleObjects/Source/configSource_inter.f90

```diff
@@ -134,7 +134,8 @@ function sampleParticle(self, rand) result(p)
     call self % sampleEnergyAngle(p, rand)
     call self % sampleEnergy(p, rand)
     p % time = ZERO
-    p % wgt  = ONE
+
+    if (p % wgt  == ZERO) p % wgt = ONE
```

Comment:
This is a strange change... What's the logic?

Reply:
It's because I was setting the particle weight in sampleEnergy, which was then being overwritten by the line 'p % wgt = ONE'. I could make a new subroutine 'sampleWeight' to do this properly, assuming that sampleEnergy is more for sampling frequency instead of weight.